runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -8195,7 +8195,9 @@ genInitObjectHeader(TR::Node * node, TR::Instruction *& iCursor, TR_OpaqueClassB
 
       // a pointer to the virtual register that will actually hold the class pointer.
       TR::Register * clzReg = classReg;
-
+      // TODO: Following approach for initializing object header for array of objects in AOT is conservative.
```

Comment:
@fjeremic Currently on Z, whenever we are loading a constant address, we use snippet in case we are generating relocatable code.  We should add relocation support for RIL type instructions which can be used to load address so we can have better AOT compiled body.

Reply:
Opened up issue : https://github.com/eclipse/openj9/issues/4391